src/Symfony/Component/Notifier/Message/SmsMessage.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Message;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Notification\Notification;
+use Symfony\Component\Notifier\Notification\SmsNotificationInterface;
+use Symfony\Component\Notifier\Receiver\Receiver;
+use Symfony\Component\Notifier\Receiver\SmsReceiverInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class SmsMessage implements MessageInterface
+{
+    private $transport;
+    private $subject;
+    private $phone;
+
+    public function __construct(string $phone, string $subject)
+    {
+        $this->subject = $subject;
+        $this->phone = $phone;
+    }
+
+    public static function fromNotification(Notification $notification, Receiver $receiver, string $transport = null): self
+    {
+        if (!$receiver instanceof SmsReceiverInterface) {
+            throw new LogicException(sprintf('To send a SMS message, "%s" should implement "%s" or the receiver should implement "%s".', get_class($notification), SmsNotificationInterface::class, SmsReceiverInterface::class));
+        }
+
+        return new self($receiver->getPhone(), $notification->getSubject());
+    }
+
+    /**
+     * @return $this
+     */
+    public function phone(string $phone)
```

Comment:
I think we should typehint `: self` instead of `@return $this` for all final classes.

Reply:
ping @nicolas-grekas

Reply:
It does not change anything AFAICS, and if we decide to remove the `final` status, we would need to fix the phpdocs as well. I would keep it as is.